### ISSUE TITLE
OLE error while importing hhc (build 5.0.215)

### ISSUE THREAD

**Issue by FransBouma** [RickStrahl/HelpBuilderReleases#36]
Hey Rick,

I tried to import my .hhc file and was greeted with: 

```
An unexpected error has occurred during the operation of Help Builder. The following error information is available: 
OLE error code 0x800706b5: The interface is unknown.
 System Error # 1426 
The error has been logged and you can now do one of the following:• Continue running the application 
• Exit the application
• Report the problem to us on our GitHub Repository
```

I don't think this is that useful info, but in case you've seen it before... It's on windows 8.1 x64, all windows updates installed. I don't have the microsoft helpcompiler installed, which might be the root cause. 

Clicking continue makes IE open a page but it's just showing a list of nested UL / LI elements, without text. 

(edit) importing a .chm has the same effect. Very strange. I'll try on another box (this is a VM) to see whether that makes a difference. Also installing the hhc compiler didn't fix it. Odd!

(edit) nope, same result. You can try it with this chm: http://www.llblgen.com/pages/PublicFilestreamer.aspx?DownloadID=314 (unblock it after download using windows explorer -> right click -> properties -> unblock). It does export a lot of the files when importing, but the toc is empty. Hopefully this gives insights in what's going on...

**Comment by RickStrahl** [comment]
@FransBouma I'll take a look at the CHM. Importing is definitely not an exact science and I had considered dropping it altogether because the imports are often less than satisfying.

I'll be able to see what crashes the import though so maybe I can at least help getting the content into Help Builder. 

Note though that once you do get it in it'll be raw HTML - not really what you want to be editing in. Preferred format is markdown.

**Comment by FransBouma** [comment]
Thanks, Rick :) I think the thing I need is the TOC imported, the rest is already HTML pages. I also tried a generated CHM from xmldocs but it crashed too, same error.

**Comment by RickStrahl** [comment]
So I took a look and ran against your CHM and got the same error you do with the EXE version of the app. Then I tried in dev mode to see what the problem is and the import just worked. Retried running as Administrator and it turns it out it works then. So, can you try running Help Builder as Admin and see if that at least gets the import working?

I'll check and see what is causing this - based on the error log (%documents%\Html Help Builder Projects\errors.txt) I suspect the problem is related to Internet Explorer automation that's used to parse the DOM tree. If so I can probably move that to the HTMLAgility pack code that I'm already using in other places.

Now in checking your imported Help file I see another problem however - the import is only importing the top level topics - all child topics are in the TOC but for some reason the child topic content is not importing. Going to look at that as well.

**Comment by RickStrahl** [comment]
So it turns out the issue is indeed related to a security exception from full IE. This is apparently a more recent change and due to the `<object>` tags used in the HHC files. I was able to switch to an internal Web browser control which does allow use of these tags which fixed the first problem (the COM error).

**Comment by FransBouma** [comment]
Running it as administrator indeed works. And indeed topics deeper into the tree are not visible. The `<li>` elements aren't closed in the TOC as helpware doesn't do that (and help compiler doesn't seem to care), not sure whether that's something to do with it....

**Comment by RickStrahl** [comment]
Frans, I updated the version 5.0.216 earlier and it has additional updates that will import your help file and bring in all of your topics. You can get the latest from: https://helpbuilder.west-wind.com/download.aspx. The issue with missing topics had to do with multiple topics refencing the same source - the solution is to not delete the underlying topics which is now disabled. If you use the new code you should be able to import all topics and you won't have to run as Admin.

I also added code to convert the existing Html content to Markdown which will go a long way to making the help content more usable. But it's one topic at a time because you have to check the content because the process gets you there only **most** of the way. This  won't fix all the issues of an HTML only help file but get you a bit closer to a native Help Builder approach. Most topics can be updated with a few simple search and replace operations.

If you want ping me on Skype tomorrow (late for me tonight) and I can give you a few suggestions to help with getting your existing content converted to a more native approach to work with Help Builder.

**Comment by FransBouma** [comment]
Thanks man, I'll try the new build and hopefully I reach you tomorrow on skype :) No time left today :/ 
Converting to markdown is great! That would make it indeed way more flexible. The HTML is a bit old at times, and likely some stuff isn't really convertable properly, but that's ok, I can convert those parts manually.

**Comment by RickStrahl** [comment]
I took a quick look. From what I can see most of the content comes over fine. You can strip headers and footers and just use the Layout template for that. The biggest issue is path fixups from the looks of it. Help Builder generates a flat file hierarchy where everything generates into a single folder. The import picks up some of relative links when it can figure it out and internally relinks those. But it doesn't work for everything. 

Curious to get a look with you to see what you can reuse and what ends up manual. 

Frankly I'm surprised how well the import actually works ;-) I haven't looked at this in a **long** time, this is very old code. I was able to do a bunch of nice updates that help make the import cleaner and adding the Markdown conversion was  nice addition as well. This will be handy for a number of other apps as well.

**Comment by RickStrahl** [comment]
Resolved in 5.0.216 and later.
